docs/design-tech/sync.md

```diff
@@ -1,89 +1,263 @@
+
+There are enough corner cases here to drown in. To handle them, we implement a design where, **rather than prevent invariant violations from occurring, the system repairs invariant violations that result from certain combinations of edits from different devices.** It does this as part of the sync routine. We found that there's generally a satisfactory way to do this.
+
+To resolve cycles, we rely on the fact that `base`, `local`, and `remote` all individually do not contain any cycles and design the resolution to preserve that. If the result of merging the file trees has a cycle, it must involve an unsynced local move and a remote move the client just pulled from the server. In particular, one of the files in the cycle must have been the subject of an unsynced local move. Our policy to resolve this is, **if we discover locally moved files which create a cycle with other files, we unmove all the locally moved files** (by setting the parents to their values in `base`).
+
+To resolve path conflicts, we rely on the fact that `base`, `local`, and `remote` all individually do not contain path conflicts, and therefore that one of the files involved in the conflict was somehow modified locally. Our policy to resolve this is, **if we discover a locally modified file whose new name conflicts with another file in the same folder, we rename the locally modified file by appending a number to the pre-extention file name.** For example, if a folder contains a new file called `notes.txt` from the server and a new local file called `notes.txt`, we rename the local file `notes-1.txt`. If there is already a `notes-1.txt`, we instead rename it `notes-2.txt`.
+
+To resolve orphaned files, we rely on the fact that an update for a file will never be pulled without the parent folder being pulled first. If the parent folder no longer exists locally, it's because the parent folder has been pruned, which only happens if the deletion of that folder has been synced. If the folder deletion has been synced, then the server has explicitly deleted all descendants of the folder, including the orphans. Because deletions can never be undone, **any updates to orphaned files can be safely ignored.**
+
+I know what you're wondering - does the order of these conflict resolutions matter? Yes it does. Any content conflicts need to be resolved before path conflicts because resolving content conflicts can create new files which can cause path conflicts. Any cycles need to be resolved before path conflicts because resolving cycles can move files which can cause path conflicts. Updates to orphaned files can be ignored at any time because no resolution policy depends on or modifies orphaned files.
+
+_Concern: we don't currently respect "Any cycles need to be resolved before path conflicts because resolving cycles can move files which can cause path conflicts." A failing test case is left as an exercise for the reader._
```

Comment:
> A failing test case is left as an exercise for the reader.
lol that's not how this works

Reply:
I actually fixed this already anyway; will update